stdlib_test.go

```diff
@@ -48,214 +49,271 @@ func TestStdlibAdapterUsage(t *testing.T) {
-		"2009/01/23 01:23:23: hello":        "ts=\"2009/01/23 01:23:23\" msg=hello\n",
-		"01:23:23: hello":                   "ts=01:23:23 msg=hello\n",
-		"2009/01/23 01:23:23.123123: hello": "ts=\"2009/01/23 01:23:23.123123\" msg=hello\n",
-		"2009/01/23 01:23:23.123123 /a/b/c/d.go:23: hello": "ts=\"2009/01/23 01:23:23.123123\" caller=/a/b/c/d.go:23 msg=hello\n",
-		"01:23:23.123123 /a/b/c/d.go:23: hello":            "ts=01:23:23.123123 caller=/a/b/c/d.go:23 msg=hello\n",
-		"2009/01/23 01:23:23 /a/b/c/d.go:23: hello":        "ts=\"2009/01/23 01:23:23\" caller=/a/b/c/d.go:23 msg=hello\n",
-		"2009/01/23 /a/b/c/d.go:23: hello":                 "ts=2009/01/23 caller=/a/b/c/d.go:23 msg=hello\n",
-		"/a/b/c/d.go:23: hello":                            "caller=/a/b/c/d.go:23 msg=hello\n",
-	} {
-		buf.Reset()
-		fmt.Fprint(writer, input)
-		if have := buf.String(); want != have {
-			t.Errorf("%q: want %#v, have %#v", input, want, have)
-		}
+func TestStdLibAdapter(t *testing.T) {
+	var testcases = []struct {
+		regexp                *regexp.Regexp
+		input                 string
+		want                  string
+		prefix                string
+		prefixJoinPrefixToMsg bool
+	}{
+		{
+			regexp: StdlibRegexpFull,
+			input:  `2009/01/23 01:23:23.123123 /a/b/c/d.go:23: hello world`,
+			want:   `ts="2009/01/23 01:23:23.123123" caller=/a/b/c/d.go:23 msg="hello world"`,
+		},
+		{
+			regexp: StdlibRegexpFull,
+			input:  `hello`,
+			want:   `msg=hello`,
+		},
+		{
+			regexp: StdlibRegexpFull,
+			input:  `2009/01/23: hello`,
+			want:   `ts=2009/01/23 msg=hello`,
+		},
+		{
+			regexp: StdlibRegexpFull,
+			input:  `2009/01/23 01:23:23: hello`,
+			want:   `ts="2009/01/23 01:23:23" msg=hello`,
+		},
+		{
+			regexp: StdlibRegexpFull,
+			input:  `01:23:23: hello`,
+			want:   `ts=01:23:23 msg=hello`,
+		},
+		{
+			regexp: StdlibRegexpFull,
+			input:  `2009/01/23 01:23:23.123123: hello`,
+			want:   `ts="2009/01/23 01:23:23.123123" msg=hello`,
+		},
+		{
+			regexp: StdlibRegexpFull,
+			input:  `2009/01/23 01:23:23.123123 /a/b/c/d.go:23: hello`,
+			want:   `ts="2009/01/23 01:23:23.123123" caller=/a/b/c/d.go:23 msg=hello`,
+		},
+		{
+			regexp: StdlibRegexpFull,
+			input:  `01:23:23.123123 /a/b/c/d.go:23: hello`,
+			want:   `ts=01:23:23.123123 caller=/a/b/c/d.go:23 msg=hello`,
+		},
+		{
+			regexp: StdlibRegexpFull,
+			input:  `2009/01/23 01:23:23 /a/b/c/d.go:23: hello`,
+			want:   `ts="2009/01/23 01:23:23" caller=/a/b/c/d.go:23 msg=hello`,
+		},
+		{
+			regexp: StdlibRegexpFull,
+			input:  `2009/01/23 /a/b/c/d.go:23: hello`,
+			want:   `ts=2009/01/23 caller=/a/b/c/d.go:23 msg=hello`,
+		},
+		{
+			regexp: StdlibRegexpFull,
+			input:  `/a/b/c/d.go:23: hello`,
+			want:   `caller=/a/b/c/d.go:23 msg=hello`,
+		},
+		{
+			regexp:                StdlibRegexpFull,
+			input:                 `some prefix hello`,
+			want:                  `msg=hello`,
+			prefix:                "some prefix ",
+			prefixJoinPrefixToMsg: false,
```

Comment:
I guess only need to specify this field when it needs to be `true`?

Reply:
Yes, removed.